// formiojs/i18n.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _default = {
  lng: 'cn',
  resources: {
    en: {
      translation: {
        complete: '提交成功',
        error: '请在提交前修正以下错误',
        required: '{{field}} 是必填项',
        pattern: '{{field}} 不匹配模式 {{pattern}}',
        minLength: '{{field}} 必须大于 {{length}} 字符。',
        maxLength: '{{field}} 必须短于 {{length}} 字符。',
        minWords: '{{field}} 必须大于 {{length}} 单词.',
        maxWords: '{{field}} 必须短于 {{length}} 单词.',
        min: '{{field}} 不能小于 {{min}}.',
        max: '{{field}} 不能大于 {{max}}.',
        maxDate: '{{field}} 不能超出日期 {{- maxDate}}',
        minDate: '{{field}} 不能早于日期 {{- minDate}}',
        invalid_email: '{{field}} 必须是有效的电子邮件',
        // eslint-disable-line camelcase
        invalid_url: '{{field}} 必须是有效的url',
        // eslint-disable-line camelcase
        invalid_regex: '{{field}} 不符合正则模式 {{regex}}.',
        // eslint-disable-line camelcase
        invalid_date: '{{field}} 不是有效日期',
        // eslint-disable-line camelcase
        invalid_day: '{{field}} 不是有效日期',
        // eslint-disable-line camelcase
        mask: '{{field}} 与掩码不匹配',
        stripe: '{{stripe}}',
        month: '月',
        day: '日',
        year: '年',
        january: '一月',
        february: '二月',
        march: '三月',
        april: '四月',
        may: '五月',
        june: '六月',
        july: '七月',
        august: '八月',
        september: '九月',
        october: '十月',
        november: '十一月',
        december: '十二月',
        next: '下一个',
        previous: '前一个',
        cancel: '取消',
        submit: '提交表单'
      }
    }
  }
};
exports.default = _default;
